git-link.el

```diff
@@ -142,10 +142,22 @@
   (save-restriction
     (widen)
     (save-excursion
-      (let ((use-region (use-region-p)))
-        (list
-         (line-number-at-pos (when use-region (region-beginning)))
-         (when use-region (line-number-at-pos (region-end))))))))
+      (let* ((use-region (use-region-p))
+             (start (when use-region (region-beginning)))
+             (end   (when use-region (region-end)))
+             (line-start (line-number-at-pos start))
+             line-end)
+        (when use-region
+          ;; If the `end' position is at the beginning of a line
+          ;; decrement the position by 1, so that the resultant
+          ;; position is on the previous line
+          (when (= end (line-beginning-position))
```

Comment:
I believe I had tested that scenario and also had read that
region-beginning and region-end automatically swap the start and end
positions such that end is greater than or equal to start.

I'll test that scenario when I get to a computer.

On Sat, May 16, 2015, 10:14 AM Skye Shaw <notifications@github.com> wrote:

> In git-link.el
> <https://github.com/sshaw/git-link/pull/13#discussion_r30461379>:
>
> > @@ -142,10 +142,22 @@
> >    (save-restriction
> >      (widen)
> >      (save-excursion
> > -      (let ((use-region (use-region-p)))
> > -        (list
> > -         (line-number-at-pos (when use-region (region-beginning)))
> > -         (when use-region (line-number-at-pos (region-end))))))))
> > +      (let* ((use-region (use-region-p))
> > +             (start (when use-region (region-beginning)))
> > +             (end   (when use-region (region-end)))
> > +             (line-start (line-number-at-pos start))
> > +             line-end)
> > +        (when use-region
> > +          ;; If the `end' position is at the beginning of a line
> > +          ;; decrement the position by 1, so that the resultant
> > +          ;; position is on the previous line
> > +          (when (= end (line-beginning-position))
>
> This does not account for the reverse scenario: the point position is less
> than the mark position. In this case the extraneous line is still included.
> For example, *P* is point, *M* is mark:
>
> 1. *P*one
> 2. two
> 3. three
> 4. *M*
>
> calling git-link with the arraignment will include lines 1-4.
>
> Need to check if (region-end) is at the beginning of the line, not (point).
> Off hand not sure how to do this without moving point. Ideas?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/sshaw/git-link/pull/13/files#r30461379>.
>